src/time.rs

```diff
@@ -161,6 +223,14 @@ impl Sub<GpsTime> for GpsTime {
     }
 }
 
+impl Sub<&GpsTime> for GpsTime {
+    type Output = Duration;
+    fn sub(self, rhs: &GpsTime) -> Duration {
+        let diff_seconds = self.diff(rhs).abs();
```

Comment:
Is there no such thing as a negative duration?  Or rather, why do we define subtraction this way?

Reply:
The Rust `Duration` type can't represent a negative duration ([relevant std docs](https://doc.rust-lang.org/stable/std/time/struct.Duration.html#method.from_secs_f64)). It seems like the standard library [decided on panicing](https://doc.rust-lang.org/stable/std/time/struct.Instant.html#method.duration_since) if a differencing operation results in a negative value.

Think this should change to match the standard library approach?

Reply:
Yes, I think we should match the semantics/invariants of the standard lib as much as possible

Reply:
Makes sense. I'll make the change in a separate PR so it's not hidden amongst mostly unrelated stuff